src/MyHealth/symptoms.ts

```diff
@@ -22,3 +45,81 @@ export const determineHealthAssessment = (
   }
   return HealthAssessment.NotAtRisk
 }
+
+type WithADate = { date: Posix }
+const compareDatesAscending = (
+  { date: dateLeft }: WithADate,
+  { date: dateRight }: WithADate,
+) => {
+  return dateLeft < dateRight ? -1 : 1
+}
+const compareDatesDescending = (
+  { date: dateLeft }: WithADate,
+  { date: dateRight }: WithADate,
+) => {
+  return dateLeft < dateRight ? 1 : -1
+}
+
+const groupLogEntriesDaily = (allEntries: SymptomLogEntry[]): LogDataByDay => {
+  const groupedEntries: LogDataByDay = {}
+
+  allEntries.forEach((entry) => {
+    const { date } = entry
+    const entryDateBeginningOfDay = beginningOfDay(date)
+
+    if (groupedEntries[entryDateBeginningOfDay]) {
+      const newLogEntries = groupedEntries[entryDateBeginningOfDay].logEntries
```

Comment:
Just curious why doesn't this need to be spread?

`const newLogEntries = [...groupedEntries[entryDateBeginningOfDay].logEntries]`

Reply:
Since we are reading a value from an object, and it is coming from an argument it doesn't get passed as a reference, this is an actual copy of it. I think it's called call by sharing or something like that 🤔 if we were to modify the actual grouped entries then we would have to spread it, but we are accessing an attribute so we get a value back

Reply:
Very cool! Thanks for the explanation.